From 707571d7df11af854dabd488523274620f8c97bd Mon Sep 17 00:00:00 2001
From: Xiaowen Liu <b37945@freescale.com>
Date: Mon, 25 Mar 2013 15:00:45 +0800
Subject: [PATCH 432/635] ENGR00242369 add correct focal length and sensitive
 element size.

fix Panorama exit auto and can't save a picture if try to
rotate the board for 360 degree.
the root cause is view angle is not right when decide to collect
the picture number.
add correct focal length and sensitive element size to calculate
the right view angle.

Signed-off-by: Xiaowen Liu <b37945@freescale.com>
---
 mx6/libcamera2/CameraUtil.h      | 2 ++
 mx6/libcamera2/MetadaManager.cpp | 4 +++-
 mx6/libcamera2/Ov5640Csi.cpp     | 4 +++-
 mx6/libcamera2/Ov5640Mipi.cpp    | 4 +++-
 mx6/libcamera2/Ov5642Csi.cpp     | 4 +++-
 5 files changed, 14 insertions(+), 4 deletions(-)

diff --git a/mx6/libcamera2/CameraUtil.h b/mx6/libcamera2/CameraUtil.h
index 40aea78..884936c 100755
--- a/mx6/libcamera2/CameraUtil.h
+++ b/mx6/libcamera2/CameraUtil.h
@@ -138,6 +138,8 @@ struct SensorInfo
     int mTargetFpsRange[MAX_FPS_RANGE];
     int mMaxWidth;
     int mMaxHeight;
+    float mPhysicalWidth;
+    float mPhysicalHeight;
     float mFocalLength;
 };
 
diff --git a/mx6/libcamera2/MetadaManager.cpp b/mx6/libcamera2/MetadaManager.cpp
index b5b527d..c154f5b 100755
--- a/mx6/libcamera2/MetadaManager.cpp
+++ b/mx6/libcamera2/MetadaManager.cpp
@@ -723,7 +723,9 @@ status_t MetadaManager::createStaticInfo(camera_metadata_t **info, bool sizeRequ
     ADD_OR_SIZE(ANDROID_SENSOR_COLOR_FILTER_ARRANGEMENT,
             &kColorFilterArrangement, 1);
 #endif
-    static const float sensorPhysicalSize[2] = {3.20f, 2.40f}; // mm
+    static const float sensorPhysicalSize[2] =
+        {mSensorInfo->mPhysicalWidth, mSensorInfo->mPhysicalHeight
+    }; // mm
     ADD_OR_SIZE(ANDROID_SENSOR_PHYSICAL_SIZE,
             sensorPhysicalSize, 2);
 
diff --git a/mx6/libcamera2/Ov5640Csi.cpp b/mx6/libcamera2/Ov5640Csi.cpp
index 873c06c..2db417b 100755
--- a/mx6/libcamera2/Ov5640Csi.cpp
+++ b/mx6/libcamera2/Ov5640Csi.cpp
@@ -137,7 +137,9 @@ status_t Ov5640Csi::initSensorInfo(const CameraInfo& info)
 
     setMaxPictureResolutions();
     FLOGI("mMaxWidth:%d, mMaxHeight:%d", mMaxWidth, mMaxHeight);
-    mFocalLength = 10.001;
+    mFocalLength = 3.42f;
+    mPhysicalWidth = 3.673f;
+    mPhysicalHeight = 2.738f;
 
     return NO_ERROR;
 }
diff --git a/mx6/libcamera2/Ov5640Mipi.cpp b/mx6/libcamera2/Ov5640Mipi.cpp
index 81e4398..681605b 100755
--- a/mx6/libcamera2/Ov5640Mipi.cpp
+++ b/mx6/libcamera2/Ov5640Mipi.cpp
@@ -131,7 +131,9 @@ status_t Ov5640Mipi::initSensorInfo(const CameraInfo& info)
 
     setMaxPictureResolutions();
     FLOGI("mMaxWidth:%d, mMaxHeight:%d", mMaxWidth, mMaxHeight);
-    mFocalLength = 10.001;
+    mFocalLength = 3.42f;
+    mPhysicalWidth = 3.673f;
+    mPhysicalHeight = 2.738f;
 
     return NO_ERROR;
 }
diff --git a/mx6/libcamera2/Ov5642Csi.cpp b/mx6/libcamera2/Ov5642Csi.cpp
index a3b756a..cc9c58d 100755
--- a/mx6/libcamera2/Ov5642Csi.cpp
+++ b/mx6/libcamera2/Ov5642Csi.cpp
@@ -130,7 +130,9 @@ status_t Ov5642Csi::initSensorInfo(const CameraInfo& info)
 
     setMaxPictureResolutions();
     FLOGI("mMaxWidth:%d, mMaxHeight:%d", mMaxWidth, mMaxHeight);
-    mFocalLength = 10.001;
+    mFocalLength = 3.34f;
+    mPhysicalWidth = 3.673f;
+    mPhysicalHeight = 2.738f;
 
     return NO_ERROR;
 }
-- 
1.8.0

